### PR TITLE
Accent Project

### PR DESCRIPTION
**Pull request by Hopekz**
A lot of work went into this PR over a long period of time.

**Adds 20 accents into the game along with an accent framework**
------
EDIT: modified as it just changed.
image
image

**Q: What is it?**
A: This adds accents into the game select-able in character preferences plus a randomized accent negative trait that gives you a random accent.

**Q: What is an accent?**
A: An accent modifies what your character says to give you a particular "accent". 

Example:
------
For example, an owo accent would turn 
**"Can I get insulated gloves, please?"**
into
**"Can i get insuwated gwoves~ pwease?"**

if you want more examples just look over the Files changed or wait until I make more.

Accent list:
-----
Please keep in mind that all these accents are unique. 
**For example, light German and Kraut don't have anything to do with one another as I have designed both of these from the ground up.**
The only exception to this is heavy and light Shakespearean as the light variant is just easier to understand.

- Belter
- Brooklyn
- Connery
- Cowboy
- Light Cowboy
- heavy French
- Kraut
- German
- Jive
- old English
- ork
- owo
- Pirate
- light Pirate
- Scottish
- light Scottish
- light Shakespearean
- heavy Shakespearean
- Muppet 

Muppet Reference: 
https://www.youtube.com/watch?v=B7UmUX68KtE
https://www.youtube.com/watch?v=_8SGRm3JaW8

- Valley

### TODO:

- [x] Fix accent trait to give you a list to choose from
- [x] fix random accent trait to successfully pick a random accent
- [x] Pay Altoids to finish the framework I started

**In the Future:**
- re-do every single accent to work with a new wordlist that was generated via server say logs (this will mean that things people actually say "in-game" would be translated rather than the topmost used 1000 dictionary words
- revamp existing accents. For example owo and valley as I have lost about 200 of my original translations. (Really noticeable in Valley)

#### Changelog

:cl:  Hopek
rscadd: Added a framework to support accents.
rscadd: Added a new negative trait to pick a random accent.
rscadd: Added 20 new unique accents to our glorious game!
/:cl:

### CODE REVIEW

**Comment by Readystorm546** [comment]
Im a simple man. I see the fact that you can UwU and ill upvote. (Ironically)

**Comment by boodaliboo** [comment]
Please add a accent for me

**Comment by boodaliboo** [comment]
Also please make owo accent admin only I really don’t want to see that ingame

**Comment by Szyszkrzyneczka** [comment]
Ruskie accent when

File: code/datums/traits/good.dm [hunk 1]
```diff
@@ -183,4 +183,68 @@
...
 	lose_text = "<span class='danger'>You no longer feel HONGRY.</span>"
-	medical_record_text = "Patient demonstrates a disturbing capacity for eating."
\ No newline at end of file
+	medical_record_text = "Patient demonstrates a disturbing capacity for eating."
+
+/datum/quirk/accent
+	name = "Accent"
+	desc = "You have developed an accent."
+	value = -1
+	mob_trait = TRAIT_RANDOM_ACCENT
```
Comment (Bibby0110) [inline code/datums/traits/good.dm]: mob_trait = TRAIT_ACCENT

File: code/datums/traits/good.dm [hunk 2]
```diff
@@ -183,4 +183,68 @@
...
 	gain_text = "<span class='notice'>You feel HONGRY.</span>"
 	lose_text = "<span class='danger'>You no longer feel HONGRY.</span>"
-	medical_record_text = "Patient demonstrates a disturbing capacity for eating."
\ No newline at end of file
+	medical_record_text = "Patient demonstrates a disturbing capacity for eating."
+
+/datum/quirk/accent
+	name = "Accent"
+	desc = "You have developed an accent."
+	value = -1
```
Comment (Bibby0110) [inline code/datums/traits/good.dm]: value = 0

**Comment by Hopekz** [comment]
> Please add a accent for me

> Ruskie accent when

This accent framework makes it easy for other people to PR their own accents as well.. so you can add them yourself I guess. 
I already made 20 accents.

File: code/datums/traits/good.dm [hunk 3]
```diff
@@ -183,4 +183,68 @@
...
+		"heavy Shakespearean",
+		"heavy Swedish",
+		"Valley")
+	var/choice = input(quirk_holder,"Which accent do you have?","Select an accent") as null|anything in accentlist
+	var/list/accent_names = strings("accents.json", "accent_file_names", directory = "strings/accents")
+
+	
+
+		 
+	//var/RNG_accent = pick("Belter") // placeholder. Doesn't even do anything at the moment.
```
Comment (Bibby0110) [inline code/datums/traits/good.dm]: Remove

File: code/datums/traits/good.dm [hunk 4]
```diff
@@ -183,4 +183,68 @@
...
+
+/datum/quirk/accent/proc/generate_accent_names()
```
Comment (Bibby0110) [inline code/datums/traits/good.dm]: Does nothing

File: code/datums/traits/good.dm [hunk 5]
```diff
@@ -183,4 +183,68 @@
...
+		message = " [message]"
+		var/list/accent_words = strings("accent_valspeak.json", "Valley", directory = "strings/accents")
+		for(var/key in accent_words)
+			var/value = accent_words[key]
+			if(islist(value))
+				value = pick(value)
+			message = replacetextEx(message, " [uppertext(key)]", " [uppertext(value)]")
+			message = replacetextEx(message, " [capitalize(key)]", " [capitalize(value)]")
+			message = replacetextEx(message, " [key]", " [value]")
+		//if(prob(100))
```
Comment (Bibby0110) [inline code/datums/traits/good.dm]: test

File: code/datums/traits/good.dm [hunk 6]
```diff
@@ -183,4 +183,68 @@
...
+	//var/RNG_accent = pick("Belter") // placeholder. Doesn't even do anything at the moment.
+
+/datum/quirk/accent/proc/generate_accent_names()
+	return list()
+
+/datum/quirk/accent/proc/handle_speech(datum/source, mob/speech_args)
+	var/message = speech_args[SPEECH_MESSAGE]
+	if(message[1] != "*")
+		message = " [message]"
+		var/list/accent_words = strings("accent_valspeak.json", "Valley", directory = "strings/accents")
```
Comment (Bibby0110) [inline code/datums/traits/good.dm]: Performance? Cache?

File: code/datums/traits/negative.dm [hunk 7]
```diff
@@ -576,3 +576,38 @@
 	H.remove_language(/datum/language/common)
 	if(!H.can_speak_in_language(/datum/language/draconic) && !H.can_speak_in_language(/datum/language/machine))
 		H.grant_language(/datum/language/japanese)
+
+/datum/quirk/random_accent
+	name = "random Accent"
```
Comment (Bibby0110) [inline code/datums/traits/negative.dm]: name = "Random Accent"

File: code/datums/traits/negative.dm [hunk 8]
```diff
@@ -576,3 +576,38 @@
...
+		"owo",
+		"Pirate",
+		"light Pirate",
+		"Scottish",
+		"light Scottish",
+		"light Shakespearean",
+		"heavy Shakespearean",
+		"heavy Swedish",
+		"Valley")
+		//H.remove_language(/datum/language/common)
```
Comment (Bibby0110) [inline code/datums/traits/negative.dm]: Remove?

File: code/datums/traits/negative.dm [hunk 9]
```diff
@@ -576,3 +576,38 @@
 	H.remove_language(/datum/language/common)
 	if(!H.can_speak_in_language(/datum/language/draconic) && !H.can_speak_in_language(/datum/language/machine))
 		H.grant_language(/datum/language/japanese)
+
+/datum/quirk/random_accent
```
Comment (Bibby0110) [inline code/datums/traits/negative.dm:582]: Perhaps you can make this a child of the /datum/quirk/accent? Someone else comment on this

**Review by Bibby0110 — COMMENTED** [review]
Some helpful comments, most of them can be ignored, good luck

File: code/datums/traits/good.dm (hunk 1 above)
Comment (Hopekz) [reply code/datums/traits/good.dm]: Sorry, I'm keeping this for debug testing. 
It's a WIP. 
Why are you even reviewing this right now its clearly a WIP? Learn to read.

File: code/datums/traits/good.dm (hunk 3 above)
Comment (Hopekz) [reply code/datums/traits/good.dm]: Sorry, I'm keeping this for debug testing. 
It's a WIP. 
Why are you even reviewing this right now its clearly a WIP? Learn to read.

File: code/datums/traits/good.dm (hunk 4 above)
Comment (Hopekz) [reply code/datums/traits/good.dm]: It's a WIP. 
It doesn't have a function yet.....
Why are you even reviewing this right now its clearly a WIP? Learn to read.

File: code/datums/traits/good.dm (hunk 3 above)
Comment (Bibby0110) [reply code/datums/traits/good.dm]: Because I like you

File: code/datums/traits/good.dm (hunk 5 above)
Comment (Hopekz) [reply code/datums/traits/good.dm]: test confirmed. You did a review

File: code/datums/traits/good.dm (hunk 6 above)
Comment (Hopekz) [reply code/datums/traits/good.dm]: Are these questions or a review?

File: code/datums/traits/negative.dm (hunk 7 above)
Comment (Hopekz) [reply code/datums/traits/negative.dm]: Good catch friend (:

File: code/datums/traits/good.dm (hunk 5 above)
Comment (Bibby0110) [reply code/datums/traits/good.dm]: No it's a test case, remove it when you're not WIP dumbo

File: code/datums/traits/negative.dm (hunk 8 above)
Comment (Hopekz) [reply code/datums/traits/negative.dm]: Sorry, I'm keeping this for debug testing. 
It's a WIP. 
Why are you even reviewing this right now its clearly a WIP? Learn to read.

File: code/datums/traits/good.dm (hunk 4 above)
Comment (Bibby0110) [reply code/datums/traits/good.dm]: Because I like you

File: code/datums/traits/negative.dm (hunk 9 above)
Comment (Hopekz) [reply code/datums/traits/negative.dm:582]: I am commenting on this

File: code/datums/traits/good.dm (hunk 1 above)
Comment (Bibby0110) [reply code/datums/traits/good.dm]: Because I like you
Also this is just plain broken and needs to be fixed, not even debug testing

File: code/datums/traits/good.dm (hunk 3 above)
Comment (Hopekz) [reply code/datums/traits/good.dm]: > Because I like you

wanna make out?

File: code/datums/traits/good.dm (hunk 4 above)
Comment (Hopekz) [reply code/datums/traits/good.dm]: > Because I like you

wanna make out?

File: code/datums/traits/good.dm (hunk 6 above)
Comment (Bibby0110) [reply code/datums/traits/good.dm]: Both
What's the performance of getting the strings from disk each time you speak?
Save the list on the datum if anything, shouldn't take up too much memory I hope

File: code/datums/traits/negative.dm (hunk 7 above)
Comment (Bibby0110) [reply code/datums/traits/negative.dm]: :)

File: code/datums/traits/negative.dm (hunk 8 above)
Comment (Bibby0110) [reply code/datums/traits/negative.dm]: Because I like you

File: code/datums/traits/negative.dm (hunk 9 above)
Comment (Bibby0110) [reply code/datums/traits/negative.dm:582]: Then do it

File: code/datums/traits/negative.dm (hunk 8 above)
Comment (Hopekz) [reply code/datums/traits/negative.dm]: > Because I like you

wanna make out?

File: code/datums/traits/good.dm (hunk 4 above)
Comment (Bibby0110) [reply code/datums/traits/good.dm]: Sure

File: code/datums/traits/good.dm (hunk 3 above)
Comment (Bibby0110) [reply code/datums/traits/good.dm]: Sure

File: code/datums/traits/negative.dm (hunk 8 above)
Comment (Bibby0110) [reply code/datums/traits/negative.dm]: Sure

File: code/datums/traits/negative.dm (hunk 9 above)
Comment (Hopekz) [reply code/datums/traits/negative.dm:582]: maybe *sniff* maybe

Comment (Bibby0110) [reply code/datums/traits/negative.dm:582]: :)

File: code/datums/traits/good.dm (hunk 2 above)
Comment (Hopekz) [reply code/datums/traits/good.dm]: I have to move this out of code/datums/traits/good.dm sometime into neutral some time.
I will do this after I finish making the trait work as I want it to.

**Comment by CMOisLing** [comment]
This seems like an awful idea +1 please make this happen

File: strings/accents/accent_belter.json [hunk 10]
```diff
@@ -0,0 +1,242 @@
+{
+
+    "Belter": {
```
Comment (nichlas0010) [inline strings/accents/accent_belter.json]: Accents shouldn't be completely incomprehensible.

**Review by Amelia0010 — CHANGES_REQUESTED** [review]
All of these are terrible. Some are completely incomprehensible (Belter), some are not even close to what they're labelled as (German), and some are just bad jokes (Swedish)

**Comment by Hopekz** [comment]
> All of these are terrible. Some are completely incomprehensible (Belter), some are not even close to what they're labelled as (German), and some are just bad jokes (Swedish)

We already have traits in the game that prevent people from talking to each other such as Sheltered so I don't see a big deal if someone were to choose to play with this accent. 

Also, Belter accent is a real accent from THE EXPANSE.

I think the german accents depict what I'm going for closely as an over-exaggerated native-born German trying to speak English for the first time.
It is very subjective but reading off some of my translations out loud I can still not see how they don't sound over-exaggerated native-born German accent.

The Swedish accent resembles the Sweedish chef puppet that I was going for:
(2 examples of this)
https://www.youtube.com/watch?v=B7UmUX68KtE
https://www.youtube.com/watch?v=_8SGRm3JaW8

**Comment by Amelia0010** [comment]
>Also, Belter accent is a real accent from THE EXPANSE.

>Belter is a real accent from The Expanse

>Belter is real - from The Expanse

\>real >from The Expanse

>Also, I think the german accents depict what I'm going for closely as an over-exaggerated native-born German trying to break English for the first time. It is very subjective but reading off some of my translations out loud I can still not see how they don't sound over-exaggerated native-born German accent.

Please pronounce Bzychiatrist. Or Detecdiffe. Or zee. Germans don't pronounce T's like D's, germans have way more pronounced Ts than englishspeakers. Germans don't pronounce Bs like Ps and vice verse. Germans don't pronounce all Vs like Fs, only some.

**Comment by overlord12345** [comment]
Where is ruski accent

**Comment by LoliconSlayer** [comment]
> I think the german accents depict what I'm going for closely as an over-exaggerated native-born German trying to speak English for the first time.
It is very subjective but reading off some of my translations out loud I can still not see how they don't sound over-exaggerated native-born German accent.

You ever wonder how "over-exaggerated" you can get before it just sounds overtly racist?
What's next, gonna make an Asian accent and replace all the Ls with Rs?

**Comment by boodaliboo** [comment]
> > I think the german accents depict what I'm going for closely as an over-exaggerated native-born German trying to speak English for the first time.
> > It is very subjective but reading off some of my translations out loud I can still not see how they don't sound over-exaggerated native-born German accent.
> 
> You ever wonder how "over-exaggerated" you can get before it just sounds overtly racist?
> What's next, gonna make an Asian accent and replace all the Ls with Rs?

yes that can be arranged

**Comment by boodaliboo** [comment]
OH GOD IM OFFENDED THEY MADE FUN OF BROOKLYN IM FROM THERE

**Comment by Mr-Space** [comment]
> 
> 
> OH GOD IM OFFENDED THEY MADE FUN OF BROOKLYN IM FROM THERE

EYYY I'M WALKIN' 'ERE

**Comment by Oakboscage** [comment]
Literally fucking racism lmao

**Comment by Firewolf34** [comment]
> All of these are terrible. Some are completely incomprehensible (Belter), some are not even close to what they're labelled as (German), and some are just bad jokes (Swedish)

@Nich you know we already have Swedish *in the game right now* as a Genetics trait. Even though it's broken as shit without Unicode support. 

@Hopekz - I love your language work. maybe there's a way to blend this with the current shitty "accents" in genetics. Have your accent code be a literal mutation, you get a random one when you're mutated. Then rip out and throw away all the old shitty accent gene mutations because nobody wants em anyways. 

Another thing is. I think this needs more IC references and less OOC. perhaps make the way lizards talk (e.g. "assss blassst usssa") a Draconic accent?

I think there's potential here for each race to have their own accent (optionally and configurable of course) and this could mesh *really well* with your work on Sheltered. 

We could give Ashwalkers their own distinct accent for example. 

Basically make this a little more RP/Immersive and integrated into the current backstory/lore rather than simply for fun/memes.

Also +1 for the dank Expanse reference.

**Comment by Firewolf34** [comment]
A good way to avoid people complaining about racism is to not use very many real races. I don't think a lot of these countries exist or are relevant out here in space anyways. That's why I think it needs more IC Lore accents. Let's come up with accents for Wizards and Syndies and shit. Imagine a nuke team showing up with a forced accent. Good luck on Stealth Ops!

**Comment by Amelia0010** [comment]
>@nich you know we already have Swedish in the game right now as a Genetics trait. Even though it's broken as shit without Unicode support.

The swedish in-game isn't as terrible, but good point, we shouldn't have two swedish accents :)

**Comment by Hopekz** [comment]
> The swedish in-game isn't as terrible, but good point, we shouldn't have two swedish accents :)

I see your point. 
Re-labeled the accent to be more accurate to what this accent is attempting to recreate.
Added my reference material to the original post if you're interested.
This is not traditional Swedish but an exaggerated chef muppet accent from the reference material.

> Germans don't pronounce T's like D's

I really like the "feel" of this current accent in-game after testing so I re-labeled it to cause less confusion for individuals that are expecting traditional German. 
This is not traditional German as you have pointed out and the way I pronounce these words out loud is probably different than other's pronunciations.

**Comment by optimumtact** [comment]
Glorious, if yogs won't I'll add it

**Comment by Hopekz** [comment]
> Glorious, if yogs won't I'll add it

Should be ready to port or pick up within a week probably.
I'm usually quite busy but I'm trying my best to finish in a timely manner.
The framework I started to make it work was finished thanks to Altoids now I just need to re-create some wordlists (owo one in particular because I fucked up and lost about 200 of my original translations). Re-creating wordlists is a huge time sink apparently!

Honestly, without Altoids offering his time via a bounty to finish my framework this project would extend out by an additional week because of the amount of no free time I have lately.

**Comment by LoliconSlayer** [comment]
How about you don't make it a negative trait as to not enable power gamers

**Comment by Hopekz** [comment]
> How about you don't make it a negative trait as to not enable power gamers

This PR has already changed since the beginning. It is now a preference in the character select screen.
The random negative trait is the only quirk remaining and during play testing I found this trait to be quite negative so I am keeping it in.

-------------------

**I noticed that the work I'm currently doing is to "revamp" word lists otherwise this project is ready to be merged.**
I can just PR my additional changes down the line otherwise everything functions the way I want it to.

**I am going to remove the WIP from this PR as it is ready to be merged and will be submitting my revamped accents as I finish making them later on.**


-----
*READY*
------

**Comment by LoliconSlayer** [comment]
A random accent is hardly an issue for a powergamer, they'll gladly be harder to understand if it means they can stack up on more positive traits when they can easily just bypass the communication issue but using their PDA, avoiding specific words, or just not talking in general.

**Comment by Hopekz** [comment]
> A random accent is hardly an issue for a powergamer, they'll gladly be harder to understand if it means they can stack up on more positive traits when they can easily just bypass the communication issue but using their PDA, avoiding specific words, or just not talking in general.

I think the value that this trait provides in-game outweighs the 1 point someone would get if they were to "powergame" around it by essentially becoming a mime. 
Going out of your way to communicate via other means is enough of a negative to warrant 1 point IMO. The goal of this trait is to either create an obstacle or create a randomized element of difficulty via these accents.

Your opinion was considered.
